common/src/main/java/com/wynntils/core/features/overlays/Overlay.java

```diff
@@ -4,24 +4,42 @@
  */
 package com.wynntils.core.features.overlays;
 
+import com.google.common.base.CaseFormat;
 import com.mojang.blaze3d.platform.Window;
 import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.features.Configurable;
+import com.wynntils.core.features.Translatable;
 import com.wynntils.core.features.overlays.sizes.FixedOverlaySize;
 import com.wynntils.core.features.overlays.sizes.OverlaySize;
 import com.wynntils.mc.render.HorizontalAlignment;
 import com.wynntils.mc.render.VerticalAlignment;
+import java.util.ArrayList;
+import java.util.List;
+import net.minecraft.client.resources.language.I18n;
 
-public abstract class Overlay {
+public abstract class Overlay implements Translatable, Configurable {
+    private List<ConfigHolder> configOptions = new ArrayList<>();
+
+    @Config
     protected OverlayPosition position;
 
+    @Config
     protected OverlaySize size;
 
+    @Config(key = "overlay.wynntils.overlay.userEnabled")
```

Comment:
What does the `key` here mean? Does that not lead to all overlays being turned on/off at the same time? Also, in terms of getting a nice looking user fronting name, would not just "enabled" be better than "userEnabled"?